test/WebSites/ControllersFromServicesWebSite/Startup.cs

```diff
@@ -25,10 +25,15 @@ public void ConfigureServices(IServiceCollection services)
                 .AddMvc()
                 .ConfigureApplicationPartManager(manager => manager.ApplicationParts.Clear())
                 .AddApplicationPart(typeof(TimeScheduleController).GetTypeInfo().Assembly)
-                .ConfigureApplicationPartManager(manager => manager.ApplicationParts.Add(new TypesPart(
-                    typeof(AnotherController),
-                    typeof(ComponentFromServicesViewComponent),
-                    typeof(InServicesTagHelper))))
+                .ConfigureApplicationPartManager(manager =>
+                {
+                    manager.ApplicationParts.Add(new TypesPart(
+                      typeof(AnotherController),
+                      typeof(ComponentFromServicesViewComponent),
+                      typeof(InServicesTagHelper)));
+
+                    manager.FeatureProviders.Add(new AssemblyMetadataReferenceFeatureProvider());
```

Comment:
This site's application manager is wonky because of this line - https://github.com/aspnet/Mvc/pull/4672/files/549d113b37a6baed6871dee4be7721da0df6301e#diff-8b7f7cd9709495478436a4d926a77252L26. Other apps don't require this.